apps/src/code-studio/components/libraries/LibraryManagerDialog.jsx

```diff
@@ -286,34 +298,89 @@ export class LibraryManagerDialog extends React.Component {
     });
   };
 
-  viewCode = library => {
+  viewCode = (library, mode) => {
     if (!library) {
       return;
     }
 
-    this.setState({viewingLibrary: library, isViewingCode: true});
+    this.setState({
+      displayLibrary: library,
+      displayLibraryMode: mode || DisplayLibraryMode.VIEW
+    });
   };
 
   closeLibraryManager = () => {
     this.setState({cachedClassLibraries: []});
     this.props.onClose();
   };
 
+  renderDisplayLibrary = () => {
+    const {displayLibrary, displayLibraryMode} = this.state;
+    if (
+      !displayLibrary ||
+      !Object.values(DisplayLibraryMode).includes(displayLibraryMode)
+    ) {
+      return;
+    }
+
+    const onClose = () =>
+      this.setState({
+        displayLibrary: null,
+        displayLibraryMode: DisplayLibraryMode.NONE
+      });
+
+    if (displayLibraryMode === DisplayLibraryMode.VIEW) {
```

Comment:
nit: slight preference for a switch statement here, which would make it explicit that you cover each case of the enum, and would make it easier to catch if we add another mode. Not blocking though!

Reply:
that makes sense to me -- i'll add `return null;` as the default case for now